newIDE/app/src/SceneEditor/index.js

```diff
@@ -1684,6 +1695,117 @@ export default class SceneEditor extends React.Component<Props, State> {
     onExtractAsExternalLayout(newName);
   };
 
+  extractAsCustomObject = (
```

Comment:
That's enough code to be moved to a util function in its own file like "CustomObjectExtractor" (and a simple test would be perfect)